Source/WebCore/platform/graphics/mac/controls/ControlMac.mm

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (C) 2022 Apple Inc. All Rights Reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import "config.h"
+#import "ControlMac.h"
+
+#if PLATFORM(MAC)
+
+#import "ColorCocoa.h"
+#import "GraphicsContextCG.h"
+#import "LocalCurrentGraphicsContext.h"
+#import "WebControlView.h"
+#import <pal/spi/cocoa/NSButtonCellSPI.h>
+#import <pal/spi/mac/NSGraphicsSPI.h>
+
+namespace WebCore {
+
+ControlMac::ControlMac(ControlPart& part, ControlFactoryMac& controlFactory)
+    : PlatformControl(part)
+    , m_controlFactory(controlFactory)
+{
+}
+
+void ControlMac::setFocusRingClipRect(const FloatRect& clipBounds)
+{
+    [WebControlView setClipBounds:clipBounds];
+}
+
+void ControlMac::updateCellStates(const FloatRect&, const ControlStyle& style)
+{
+    [WebControlWindow setHasKeyAppearance:!style.flags.contains(ControlStyle::Flag::WindowInactive)];
+    [WebControlView setUseSemanticContext:style.flags.contains(ControlStyle::Flag::WindowInactive)];
```

Comment:
Incorrect flag used here, but this line should be irrelevant if you make the other changes described.

Reply:
This line is removed and `[WebControlView setUseSemanticContext]` is also removed.